File: App/lib/social-medias/social_media.dart
class SocialMedia {
  String? id;
  String? type;
  String link;

  SocialMedia(
      {this.id,
      this.type,
      required this.link});

  static SocialMedia fromMap(Map<String, dynamic> map) {
    return SocialMedia(
        id: map['id'],
        type: map['type'],
        link: map['link']);
  }
}

File: Api/Benchmark/Contracts/SocialMedia.cs
using System.Text.Json.Serialization;

namespace Benchmark.Contracts
{
    public class SocialMedia
    {
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        [JsonPropertyName("type")]
        public string Type { get; set; }

        [JsonRequired]
        [JsonPropertyName("link")]
        public string Link { get; set; }
    }
}